### PR TITLE
Add minhash deduplicator based on RAY.

### PR DESCRIPTION
**Pull request by chenyushuo**
Unlike #489, the main approach here is based on Ray Actor's implementation of multi process union find set to complete equivalence class merging.

### CODE REVIEW

**Review by pan-x-c — COMMENTED** [review]
Please see the inline comments

File: environments/dist_requires.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
-ray==2.31.0
+ray<=2.38.0
```
[the remark this answers is not in the document]
Comment (chenyushuo) [reply environments/dist_requires.txt]: ok

File: data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+                self.union_threshold,
+                self.union_find_parallel_num,
+                i,
+                self.remote_edge_buffers,
+                self.max_pending_edge_buffer_task,
+                self.num_edge_buffer_task_returns,
+            ) for i in range(self.union_find_parallel_num)
+        ]
+
+        self.tmp_file_name = os.path.join(os.getcwd(), tmp_file_name,
```
Comment (pan-x-c) [inline data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py:435]: Moving the temporary file to a sub-directory of the output directory maybe better.

File: data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+            return new_table
+
+        dataset.map_batches(
+            minhash_with_uid,
+            batch_format='pyarrow',
+            zero_copy_batch=True,
+        ).write_parquet(self.tmp_file_name, force_ascii=False)
+        dataset = ray.data.read_parquet(self.tmp_file_name)
+        end_time = time.time()
+        print(f'MinHash time = {end_time - start_time}')
```
Comment (pan-x-c) [inline data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py:564]: -> `logger.info`

File: data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+        return [idx for uid, idx in queries if uid in self.parent]
+
+
+OP_NAME = 'ray_bts_minhash_deduplicator'
+
+
+@OPERATORS.register_module(OP_NAME)
+class RayBTSMinhashDeduplicator(Deduplicator):
+    """
+    A basic exact matching deduplicator for RAY.
```
Comment (pan-x-c) [inline data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py:239]: Update the description.

File: data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py (hunk 3 above)
Comment (chenyushuo) [reply data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py]: fixed

File: data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py (hunk 4 above)
Comment (chenyushuo) [reply data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py]: fixed

File: data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py (hunk 2 above)
Comment (chenyushuo) [reply data_juicer/ops/deduplicator/ray_bts_minhash_deduplicator.py]: fixed

**Review by pan-x-c — APPROVED** [review]
LGTM

**Pull request merged**